### ISSUE TITLE
AbsentOrWrongFileLicense doesn't take into account test files

### ISSUE THREAD

**Issue by krzema12** [detekt/detekt#3777]
## Expected Behavior
[AbsentOrWrongFileLicense](https://detekt.github.io/detekt/comments.html#absentorwrongfilelicense) rule should take into account test files (`src/test/kotlin/**/*.kt`).

## Observed Behavior
The rule doesn't detect missing license preamble even if I explicitly call `detektTest` (I'm aware of https://github.com/detekt/detekt/issues/2956).

## Steps to Reproduce

Custom config:

```
comments:
  AbsentOrWrongFileLicense:
    active: true
    licenseTemplateFile: license.template
```

Gradle config:
```
detekt {
    buildUponDefaultConfig = true
    config = files("$projectDir/detekt-custom-config.yml")
}
```

I'll publish a link to my open-source project once the project is public :)

## Context

I'm working on covering legal requirements to publish an open-source library from my company. I found it when adding this preamble to existing files, and Detekt didn't complain about missing preamble in some test file.

This bug can have potentially legal implications.

## Your Environment

* Version of detekt used: 1.16.0
* Version of Gradle used (if applicable): 6.8
* Operating System and version: Ubuntu 20.04.2 LTS
* Link to your project (if it's a public repository): will be public within several days

**Comment by zmunm** [comment]
Hi. By default, [detekt does not contain test sourceSet](https://github.com/detekt/detekt/blob/main/detekt-core/src/main/resources/default-detekt-config.yml#L51), [comment rules exclude test folder](https://github.com/detekt/detekt/blob/main/detekt-core/src/main/resources/default-detekt-config.yml#L51).

So, to achieve this, modify it like this:

config
```yml
comments:
  excludes: []
  AbsentOrWrongFileLicense:
    active: true
    licenseTemplateFile: license.template
```

gradle
```kotlin
detekt {
    buildUponDefaultConfig = true
    config = files("$projectDir/detekt-custom-config.yml")
    input = files(
        "$projectDir/src/main/",
        "$projectDir/src/test/",
    )
}
```

**Comment by BraisGabin** [comment]
Running detektTest and the configuration pointed by @zmunm should work too. And the last version of detekt includes test source set by default (the current RC) so even the Detekt task will include those by default.

Anyway, we should probably remove that exclude for this rule...

**Comment by krzema12** [comment]
Thanks, adding `excludes: []` worked. However I agree with @BraisGabin it would be safer to not exclude anything in this rule. I was close to not spotting this issue, which worst case might have legal consequences for my company.
